lib/commands/help.js

```diff
@@ -93,24 +94,26 @@ module.exports = Command.extend({
   },
 
   _printJsonHelp: function(commandOptions, rawArgs) {
-    var generator = new JsonGenerator({
-      ui: this.ui,
-      project: this.project,
-      commands: this.commands,
-      tasks: this.tasks
-    });
 
-    var json = generator.generate(commandOptions, rawArgs);
+    var jsonString;
+    var generateJsonCommand = this._lookupCommand('generateJson');
+    var cachePath = this.project.root +
+                    generateJsonCommand.cacheDirectory + '/' +
+                    generateJsonCommand.cacheFileName;
 
-    var outputJsonString = JSON.stringify(json, function(key, value) {
-      // build command has a recursive property
-      if (value === path) {
-        return 'path';
-      }
-      return value;
-    }, 2);
+    var hasCache = fs.existsSync(cachePath);
+
+    if (rawArgs.length === 0 && hasCache) {
+
+      jsonString = fs.readFileSync(cachePath, 'utf8');
+
+    } else {
+
+      jsonString = generateJsonCommand.run(commandOptions, rawArgs);
```

Comment:
sounds reasonable, this will also address most of the comments you had inside the generate-json command (it was effectively handling args and commands for this very case here.